src/partitioning.c

```diff
@@ -26,18 +26,33 @@
 #include "catalog.h"
 #include "utils.h"
 
-#define IS_VALID_PARTITIONING_FUNC(proform)							\
-	((proform)->prorettype == INT4OID &&							\
-	 ((proform)->provolatile == PROVOLATILE_IMMUTABLE) &&			\
+#define IS_VALID_CLOSED_PARTITIONING_FUNC(proform, argtype)				\
+	((proform)->prorettype == INT4OID &&								\
+	 ((proform)->provolatile == PROVOLATILE_IMMUTABLE) &&				\
+	 (proform)->pronargs == 1 &&										\
+	 ((proform)->proargtypes.values[0] == argtype ||					\
+	  (proform)->proargtypes.values[0] == ANYELEMENTOID))
+
+#define IS_VALID_OPEN_PARTITIONING_FUNC(proform, argtype)			\
+	(IS_VALID_OPEN_DIM_TYPE((proform)->prorettype) &&				\
 	 (proform)->pronargs == 1 &&									\
-	 (proform)->proargtypes.values[0] == ANYELEMENTOID)
-/*
- * Get the OID of the partitioning function given a qualified name.
- *
- * Only functions that match the supported function signature will be returned.
- */
+	 ((proform)->provolatile == PROVOLATILE_IMMUTABLE) &&			\
+	 ((proform)->proargtypes.values[0] == argtype ||				\
+	  (proform)->proargtypes.values[0] == ANYELEMENTOID))
+
+#define IS_VALID_PARTITIONING_FUNC(proform, dimtype, argtype) \
+	((dimtype == DIMENSION_TYPE_OPEN) ?						  \
+	 IS_VALID_OPEN_PARTITIONING_FUNC(proform, argtype) :	  \
+	 IS_VALID_CLOSED_PARTITIONING_FUNC(proform, argtype))
+
+typedef bool (*proc_filter) (Form_pg_proc form, Oid argtype);
+
 static regproc
-partitioning_func_get(const char *schema, const char *funcname)
+lookup_proc_filtered(const char *schema,
+					 const char *funcname,
+					 Oid coltype,
```

Comment:
`coltype` is only used with filter and not checked otherwise. Should it check the argument is correct if `coltype` is a valid Oid, or just document the current behavior?

Reply:
Happy to document, but I think that from the perspective of this function it doesn't know of these details. It just applies the filter in a "dumb" way.